src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -796,6 +817,215 @@ void Transformations::MainSnippets(void) {
     };
 #endif // OPENVINO_ARCH_X86_64
 
+    const bool is_support_op = [](const std::shared_ptr<const ov::Node> &n) -> bool {
+        auto is_supported_matmul = [](const std::shared_ptr<const ov::Node>& n) -> bool {
```

Comment:
These lambda functions are support conditions for common Snippets library.
Snippets pipeline supports different nodes. However, plugin can support them only in specific pattern. For example, x64 supports MatMul, Transpose and Softmax only as part on MHA pattern.
No need to copy from common tokenization pass to this callback. If we change something in common logic, we should align callback with tokenization behavior. It's not safe.

At the moment, x64 supports only Eltwise+FQ+Convert ops in common tokenization pass. MHA pattern (MatMul, Transpose, Softmax) is tokenized by the separate MHATokenization pass. Because of that SnippetsTokenization callbacks contains these [checks](https://github.com/openvinotoolkit/openvino/blob/77635530f21b64baf42b1c3bc9bc759839d71a4a/src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp#L1062).

I'd expect here something like:

```
const bool is_support_op = [](const std::shared_ptr<const ov::Node> &n) -> bool {
#if defined(OPENVINO_ARCH_ARM64)
         return  (ov::is_type<const ov::op::v1::Add>(n) ||
                          ov::is_type<const ov::op::v8::Mul>(n) ||
                          ....);
#else
     // CPU Plugin support Swish in Subgraph via conversion to SwichCPU which assumes second input to be constant
      auto is_unsupported_swish= [](n) {
           return ov::is_type<const ov::op::v4::Swish>(n) && n->inputs().size() > 1 &&
                      !ov::is_type<const ov::op::v0::Constant>(n->get_input_node_shared_ptr(1));
      };
      / todo: general tokenization flow is not currently supported for these operations.
       //  they can be tokenized only as a part of complex patterns
      auto is_unsupported_by_common_tokenization = [](n) {
            return  (ov::is_type<const ov::op::v1::Softmax>(n) ||
                          ov::is_type<const ov::op::v8::Softmax>(n) ||
                          ov::is_type<const ov::op::v0::MatMul>(n) ||
                          ov::is_type<const ov::op::v1::Transpose>(n) ||
                          ov::is_type<const ov::op::v1::Broadcast>(n) ||
                          ov::is_type<const ov::op::v3::Broadcast>(n) ||
                          ov::is_type<const ov::op::v1::ReduceMax>(n) ||
                          ov::is_type<const ov::op::v1::ReduceSum>(n));
        };
        return !is_unsupported_swish(n) && !is_unsupported_by_common_tokenization (n);
#endif
}
```

Reply:
Applied. Thanks Alexandra!

Reply:
Yeah, now it looks better! Thanks! 🔥